Tasks/FtpUploadV2/Tests/L0.ts

```diff
@@ -1,16 +1,129 @@
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('Input required: remotePath'), 'Should have printed: Input required: remotePath');
+        assert(tr.failed, 'task should have failed');
+
+        done();
+    });
+
+    it('check args: no clean', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoClean.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('Input required: clean'), 'Should have printed: Input required: clean');
+        assert(tr.failed, 'task should have failed');
+
+        done();
+    });
+
+    it('check args: no preservePaths', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoPreservePaths.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('Input required: preservePaths'), 'Should have printed: Input required: preservePaths');
+        assert(tr.failed, 'task should have failed');
+
+        done();
     });
 
-    after(() => {
+    it('check args: no trustSSL', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoTrustSSL.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.invokedToolCount === 0, 'should not run anything');
+        assert(tr.stdOutContained('Input required: trustSSL'), 'Should have printed: Input required: trustSSL');
+        assert(tr.failed, 'task should have failed');
+
+        done();
     });
 
-    it('Does a basic hello world test', function(done: MochaDone) {
-        // TODO - add real tests
+    it('check args: no protocol on server URL (ftp:// or ftps://)', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoProtocol.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('loc_mock_FTPNoProtocolSpecified'), 'Should have printed: loc_mock_FTPNoProtocolSpecified');
+        assert(tr.failed, 'task should have failed');
+
+        done();
+    });
+
+    it('check args: no host name on server URL', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoHostName.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('loc_mock_FTPNoHostSpecified'), 'Should have printed: loc_mock_FTPNoHostSpecified');
+        assert(tr.failed, 'task should have failed');
+
+        done();
+    });
+
+    it('check args: no files found', (done: Mocha.Done) => {
+        const tp = path.join(__dirname, 'L0NoFilesFound.js');
+        const tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+
+        assert(tr.stdOutContained('loc_mock_NoFilesFound'), 'Should have printed: loc_mock_NoFilesFound');
+        assert(tr.failed, 'task should have failed');
+
```

Comment:
We don't have a single unit test for successful execution case. Maybe we should add them? @tatyana-kostromskaya

Reply:
Added test, thanks!